foundation/foundation-configuration-store/src/main/java/com/boozallen/aissemble/configuration/service/ConfigService.java

```diff
@@ -38,14 +38,16 @@ public class ConfigService {
     public Response getProperty(@PathParam("groupName") String groupName,
                               @PathParam("propertyName") String propertyName) {
         Property property = null;
+        Exception exception = null;
         try {
              property = this.configLoader.read(groupName, propertyName);
         } catch (Exception e) {
-            logger.error(String.format("Error reading property - groupName: %s, propertyName: %s", groupName, propertyName), e);
+            exception = e;
         }
         if (property != null) {
             return Response.ok(property.toJsonString()).build();
         }
+        logger.error(String.format("Error reading property - groupName: %s, propertyName: %s", groupName, propertyName), exception);
```

Comment:
Any reason for move from the above catch statement?

Reply:
I think it's because we don't throw exception for the configuration not found case anymore but we still wanna log it.

Reply:
Could we still do that on line 44 like we had it originally?

Reply:
We made some changes to the `VaultPropertyDao's read()` function. Previously, if a property was not available in the server, the` read()` function would throw an exception. However, we need to check a potentially non-existent status property at server startup, so we modified the` read() `function to not throw an exception if the property is null. Due to this change, we also had to update the `getProperty() `function in the configuration service. Previously, `getProperty()` would throw an exception when reading a non-existent property, but now it won't. Therefore, we adjusted `getProperty()` to handle null values properly. If a property is null, it will not throw an exception, but it will still fail and display an appropriate message.

Reply:
> Previously, `getProperty()` would throw an exception when reading a non-existent property, but now it won't.

Based on what used to be on line 44, it looks like the `getProperty()` function used to just `logger.error` isntead of throw an exception. So I don't understand why we're moving that same `logger.error` call to line 50 now. The functionality appears to be the same as before except now we have a `Exception exception = null` variable.

Reply:
gotcha! previously null case will through an exception and `logger.error` will be printed, not anymore with the change. We want to print the same msg    ```
logger.error(String.format("Error reading property - groupName: %s, propertyName: %s", groupName, propertyName));
```  in case of null on line 50. Thus we move it to line 50 so that as long as the property is not null, error msg will be printed out. Revert it back + adding a logger info on Line 50 seems cleaner, and I've reverted it back

Reply:
Think we could probably remove this logic altogether if we revert back to throwing the exception as part of `VaultPropertyDao's read()` and do the exception handling in the `isFullyLoaded()` function

Reply:
Did the exception in `isFullyLoaded()` and changed it back